bokeh/command/subcommands/serve.py

```diff
@@ -246,6 +246,76 @@
 should be a cryptographically random string with at least 256 bits (32 bytes)
 of entropy. The ``bokeh secret`` command can generate new secret keys.
 
+Authentication Options
+~~~~~~~~~~~~~~~~~~~~~~
+
+The Bokeh server can be configured to only allow connections in case there is
+a properly authenticed user. This is accomplished by providing the path to
+a module that implements the necessary functions on the command line:
+
+.. code-block:: sh
+
+    bokeh serve --auth-module=/path/to/auth.py
+
+or by setting the ``BOKEH_AUTH_MODULE`` environment variable.
+
+The module must contain *one* of the following two functions that will return
+the current user (or None):
+
+.. code-block:: python
+
+    def get_user(request_handler):
+        pass
+
+    async def get_user_aync(request_handler):
+        pass
+
+The function is passed the Tornado ``RequestHandler`` and can inspect cookies
+or request headers to determine the authenticated user. If there is no valid
+authenticated user, these functions should return None.
+
+Additionally, the module must specify where to redirect unauthenticated users.
+It must contain either:
+
+* a module attribute ``login_url`` and (optionally) a ``LoginHandler`` class
+* a function definition for ``get_login_url``
+
+.. code-block:: python
+
+    login_url = "..."
+
+    class LoginHandler(RequestHandler):
+        pass
+
+    def get_login_url(request_handler):
+        pass
+
+When a relative ``login_url`` is given, an optional ``LoginHandler`` class may
+also be provided, and it will be installed as a route on the Bokeh server
+automatically.
+
+The ``get_login_url`` function is useful in cases where the login URL must
+vary based on the request, or cookies, etc. It is not possible to specify a
+``LoginHandler`` when ``get_url_function`` is defined.
+
+Analogous to the login options, optional ``logout_url`` and ``LogoutHandler``
+values may be define and endopoint for logging users out.
+
+If no auth module is provided, then a default user will be assumed, and no
+authentication will be required to access Bokeh server endpoints.
+
+.. warning::
+    The contents of the auth module will be executed!
```

Comment:
Can you explain the implications here, i.e. why that's a warning? Maybe:

```suggestion
    The contents of the auth module will be executed, so you should ensure that this module is trustworthy and only includes code you are comfortable with.
```